windows/devices/uwb/UwbCxAdapter.cxx

```diff
@@ -0,0 +1,64 @@
+
+#include <algorithm>
+
+#include <windows/devices/uwb/UwbCxAdapter.hxx>
+
+#include <windows/devices/uwb/UwbAppConfiguration.hxx>
+
+using namespace windows::devices;
+
+UwbSetAppConfigurationParameters
+windows::devices::uwb::GenerateUwbSetAppConfigParameterDdi(const ::uwb::protocol::fira::UwbSessionData& uwbSessionData)
+{
+    std::vector<std::shared_ptr<IUwbAppConfigurationParameter>> parameters;
+
+    // TODO NUMBER_OF_CONTROLEES
+    // TODO DEVICE_MAC_ADDRESS
+    // TODO DST_MAC_ADDRESS
+    // TODO DEVICE_TYPE
+    const auto sessionUwbMap = uwbSessionData.uwbConfiguration.GetValueMap();
+
+    for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
+        std::visit([&](auto&& parameterValue) {
+            using ParameterValueT = std::decay_t<decltype(parameterValue)>;
+            if constexpr (std::is_enum_v<ParameterValueT>) {
+                parameters.push_back(std::make_shared<UwbAppConfigurationParameter<std::underlying_type_t<ParameterValueT>>>(notstd::to_underlying(parameterValue), parameterTag));
```

Comment:
Is there a reason why this is shared_ptr?

Reply:
Yes. `UwbAppConfigurationParameter` is a template class, which means it's not a real class; each `PropertyT` type it's instantiated with creates a unique type. Therefore, we cannot store ``UwbAppConfigurationParameter` objects in a container. So to alleviate that, we introduced a _non-templated_ base class `IUwbAppConfigurationParameter` that `UwbAppConfigurationParameter` derives from. We can store instances of `IUwbAppConfigurationParameter` in a container.

However, if we stored the base class directly in the container, then when we attempt to access the actual derived implementation, it'll fail due to object slicing. However, polymorphic access works through base class _pointers_., so we allocated and access the objects dynamically via pointer instead.

Any time you're doing dynamic allocation, one of the smart pointer types should be used to manage the allocated memory safely: `unique_ptr` if used in isolation, otherwise `shared_ptr` if it's passed around. In this case, the object is passed around, so `shared_ptr` was used.

Reply:
Ah I see, the interface is so that we don't have to use the Builder model. I guess I'm having difficulty differentiating use cases where objects are owned by several different things vs not. In this case, it appears that first the parameters vector owns the pointers, but then all the pointers are passed into the constructor of UwbSetAppConfigurationParameters. Thus it appears to me that at any point in time they are only owned by one thing.

Reply:
Actually yes, I think you're right about this, so probably should be `std::unique_ptr` here instead. I had been toying with something else where these were being passed around, but that's certainly not the case right now.

Reply:
I switched to using unique_ptr